src/configparser.cpp

```diff
@@ -108,6 +110,7 @@ bool ConfigParser::parse(const std::string& filename, bool double_include)
 			}
 		}
 	}
+	included_files.pop_back();
```

Comment:
I think this opens a possibility for double-inclusion, e.g.:
* A includes B and C;
* B includes X;
* C includes X.

On first glance, It appears to me that both an inclusion stack and a set of already processed files are needed here.

Reply:
Yeah, it does. 

If a user were to manually include the same file twice, should that be allowed?

I'm not familiar with how many problems double-including would have, while I assumed the real threat here was a recursive-include

Reply:
Current implementation disallows that, and since I don't know the rationale, I'd err on the side of caution and preserve that behaviour.

I think the only problem with multiple includes is side effects: if there is backticks with some command that e.g. changes a file, it will be executed multiple times.

I don't have time to do this right now, but if you're interested, what I'd do is open `git log --reverse` and search for `"include"` or `included_files` to trace the history of this feature.

Reply:
fixed in 4bd6f29, but see main discussion for comments